fish-rust/src/common.rs

```diff
@@ -1827,22 +1779,15 @@ pub fn scoped_push<Context, Accessor, T>(
     mut ctx: Context,
     accessor: Accessor,
     new_value: T,
-) -> impl Deref<Target = Context> + DerefMut<Target = Context>
+) -> impl DerefMut<Target = Context>
```

Comment:
I'm nitpicking but I just want to point out that the change to the inline hints that this PR brings about (which I understand was the motivation for this) could have been brought with just this single line change.

Reply:
Yeah this could have been a separate commit